addons/binding/org.openhab.binding.somfytahoma/src/main/java/org/openhab/binding/somfytahoma/handler/SomfyTahomaBridgeHandler.java

```diff
@@ -339,254 +280,151 @@ private String getFormattedParameters(Collection<String> stateNames) {
-                    if (channelState != null) {
-                        updateState(channel.getUID(), channelState);
-                    } else {
-                        logger.debug("Cannot find state for channel {}", channel.getUID());
-                    }
+                if (handler != null) {
+                    // update thing status
+                    handler.updateThingStatus(states);
+                    handler.updateThingChannels(states);
                 }
+            } else {
+                logger.debug("Thing handler is null, probably not bound thing.");
             }
-        }
-    }
 
-    private void updateThingStatus(SomfyTahomaBaseThingHandler handler, List<SomfyTahomaState> states) {
-        for (SomfyTahomaState state : states) {
-            if (state.getName().equals(STATUS_STATE) && state.getType() == TYPE_STRING) {
-                if (state.getValue().equals(UNAVAILABLE)) {
-                    handler.setUnavailable();
-                } else {
-                    handler.setAvailable();
-                }
-                return;
-            }
         }
     }
 
-    private State getChannelState(SomfyTahomaThingHandler handler, Channel channel,
-                                  List<SomfyTahomaState> channelStates) {
-        ChannelTypeUID channelUID = channel.getChannelTypeUID();
-        if (channelUID == null) {
-            return null;
-        }
-
-        String stateName = handler.getStateNames().get(channelUID.getId());
-        for (SomfyTahomaState state : channelStates) {
-            if (state.getName().equals(stateName)) {
-                return parseTahomaState(state);
+    private Thing getThingByDeviceUrl(String deviceUrl) {
+        for (Thing th : getThing().getThings()) {
+            String url = (String) th.getConfiguration().get("url");
+            if (deviceUrl.equals(url)) {
+                return th;
             }
         }
         return null;
     }
 
-    public void updateChannelState(SomfyTahomaThingHandler handler, ChannelUID channelUID, String url) {
-        if (handler.getStateNames() != null) {
-            String stateName = handler.getStateNames().get(channelUID.getId());
-            State state = getState(handler, url, stateName);
-            if (state == null || state.equals(UnDefType.UNDEF)) {
-                return;
-            }
-
-            if (state.equals(UnDefType.NULL)) {
-                // relogin
-                login();
-                state = getState(handler, url, stateName);
-            }
-
-            updateState(channelUID, state);
-        }
-    }
-
     private void logout() {
         try {
             sendToTahomaWithCookie(TAHOMA_URL + "logout");
-            cookie = "";
-        } catch (Exception e) {
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
             logger.error("Cannot send logout command!", e);
```

Comment:
Does this really hurt if the logout is not successful? Otherwise I would suggest to reduce it to debug and not annoy the user with this information

Reply:
OK